src/Libraries/Nop.Services/Helpers/BrowscapXmlParser.cs

```diff
@@ -101,7 +104,11 @@ private static string ToRegexp(string str)
         /// <returns>True if user agent is a crawler, otherwise - false</returns>
         public bool IsCrawler(string userAgent)
         {
-            return _crawlerUserAgentsRegexp.Any(p => Regex.IsMatch(userAgent, p));
+            // check against cache object
+            bool result;
+            if (_userAgentsCache.TryGetValue(userAgent, out result))
```

Comment:
It would be great if you come up with some decision on removing Regex.IsMatch challenges in the task where the user agents describes by templates.